File: apps/macos/Sources/TrimletCore/FFmpegProgress.swift
import Foundation

public enum FFmpegProgress {
    public static func elapsedSeconds(from text: String) -> Double? {
        var latestMicroseconds: Double?
        for line in text.split(separator: "\n") {
            let parts = line.split(separator: "=", maxSplits: 1)
            guard parts.count == 2,
                  parts[0] == "out_time_us" || parts[0] == "out_time_ms",
                  let value = Double(parts[1]) else { continue }
            latestMicroseconds = value
        }
        return latestMicroseconds.map { max(0, $0 / 1_000_000) }
    }
}

File: apps/windows/src/Trimlet.Media/FFmpegProgress.cs
namespace Trimlet.Media;

public static class FFmpegProgress
{
    public static TimeSpan? ElapsedFromBlock(string text)
    {
        long? microseconds = null;
        foreach (var rawLine in text.Split('\n'))
        {
            var line = rawLine.Trim();
            var separator = line.IndexOf('=');
            if (separator <= 0 || separator == line.Length - 1)
            {
                continue;
            }

            var key = line[..separator];
            if ((key == "out_time_us" || key == "out_time_ms") &&
                long.TryParse(line[(separator + 1)..], out var value))
            {
                microseconds = Math.Max(0, value);
            }
        }

        return microseconds is { } elapsed
            ? TimeSpan.FromTicks(elapsed * 10)
            : null;
    }
}
